src/category_theory/monoidal/subcategory.lean

```diff
@@ -149,6 +150,44 @@ symmetric_category_of_faithful (full_braided_subcategory_inclusion P)
 
 end symmetric
 
+section closed
+
+variables (P) [monoidal_closed C]
+
+/--
+A property `C → Prop` is a closed predicate if it is closed under taking internal homs
+-/
+class closed_predicate :=
+(prop_ihom' : ∀ {X Y}, P X → P Y → P ((ihom X).obj Y) . obviously)
```

Comment:
I'm not sure that this does what you think it does. Currently, `closed_predicate P : Type`, but what you probably wanted is `closed_predicate P : Prop`, which can be achieved by writing
```suggestion
class closed_predicate : Prop :=
(prop_ihom' : ∀ {X Y}, P X → P Y → P ((ihom X).obj Y) . obviously)
```
The same is true for `monoidal_predicate` above.

Reply:
`Prop` gives you proof irrelevance. If you have two proofs of `closed_predicate P` for the same `P` floating arond (this can happen for various reasons), then Lean will start to complain about typeclass diamonds if you don't make `closed_predicate` a `Prop`.